libcaf_net/test/net/web_socket/framing.cpp

```diff
@@ -0,0 +1,184 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#define CAF_SUITE net.web_socket.framing
+
+#include "caf/net/web_socket/framing.hpp"
+
+#include "net-test.hpp"
+
+using namespace caf;
+using namespace std::literals;
+
+namespace {
+
+class app_t : public net::web_socket::upper_layer {
```

Comment:
This is almost the same app_t as in client test. Only difference is in the abort call detection. I could elevate this to net-test? It seems different from the app_t in server tests, but I would really like to see only one app layer fake. Do you think it would be possible?

Reply:
In general, a bit of duplication is OK in tests. However, if this exact app is used in multiple tests, feel free to move it to separate hpp/cpp files as a followup.